index-set-app/src/cmr/index_set/services/index_service.clj

```diff
@@ -229,50 +228,110 @@
+  (let [individual-index-settings (get-in index-set [:index-set :granule :individual-index-settings])]
     (update-in index-set [:index-set :granule :indexes]
                conj
                {:name collection-concept-id
                 :settings individual-index-settings})))
 
+(defn- remove-granule-index-from-index-set
+  "Removes the separate granule index for the given collection from the index set. Validates the
+  collection index is listed in the index-set."
+  [index-set collection-concept-id]
+  (let [existing-index-names (->> (get-in index-set [:index-set :granule :indexes]) (map :name) set)]
+    (when-not (contains? existing-index-names collection-concept-id)
+      (errors/throw-service-error
+       :bad-request
+       (format
+         "The collection does not have a separate granule index"
+         collection-concept-id)))
+    (update-in index-set [:index-set :granule :indexes]
+               (fn [indexes]
+                 (remove #(= collection-concept-id (:name %))
+                         indexes)))))
+
 (defn mark-collection-as-rebalancing
   "Marks the given collection as rebalancing in the index set."
-  [context index-set-id concept-id]
-  (let [index-set (get-index-set context index-set-id)
-        ;; Add the collection to the list of rebalancing collections. Also
-        ;; does validation.
-        index-set (update-in
-                    index-set
-                    [:index-set :granule :rebalancing-collections]
-                    add-rebalancing-collection concept-id)
-        index-set (add-new-granule-index index-set concept-id)]
-
-    ;; Update the index set. This will create the new collection indexes as
-    ;; needed.
+  [context index-set-id concept-id target]
+  (let [index-set (as-> (get-index-set context index-set-id) index-set
+                        (update-in
+                          index-set
+                          [:index-set :granule :rebalancing-collections]
+                          add-rebalancing-collection concept-id)
+                        (update-in
+                          index-set
+                          [:index-set :granule :rebalancing-targets]
+                          assoc concept-id target)
+                        (if (= "small-collections" target)
+                          (do
+                             (validate-granule-index-exists index-set concept-id)
+                             index-set)
+                          (add-new-granule-index index-set concept-id)))]
+    ;; Update the index set. This will create the new collection indexes as needed.
     (update-index-set context index-set)))
 
+(defn- validate-target
+  "Validates the target is one of the two allowed values."
+  [target concept-id]
+  (if (nil? target)
+    (errors/throw-service-errors
+     :bad-request
+     [(format "The index set does not contain the rebalancing collection [%s]"
+              concept-id)])
+    (when-not (or (= "small-collections" target)
```

Comment:
For this sort of thing I tend to do `(when-not (contains? #{"small-collections" "separate-index"} target)` ... I have no idea which is more efficient, nor really which should be preferred in a simple case like this :-) (most useful as the set grows ...)

Reply:
I'm kind of all over the place with this one - in other places in this same PR I used:
``` (when-not (some #{target} ["small-collections" "separate-index"]))```
I think `some` is my favorite just because this style is called out as an idiom in the clojure docs. But then `contains?` operates in constant time because it's a hash so for a large number of items that must be better than `some`. Too many choices :overwhelmed:.

Reply:
`some` is my choice because I like idiom.

Reply:
Switched to `some`.